platform/ios/src/MGLAnnotationView.h

```diff
@@ -16,6 +18,14 @@ NS_ASSUME_NONNULL_BEGIN
 - (instancetype)initWithReuseIdentifier:(nullable NSString *)reuseIdentifier;
 
 /**
+ This property will be set to the associated annotation when the view is visible.
+ 
+ When the view is queued and waiting to be reused, the value will be set to nil.
+ This property should not be changed directly.
+ */
+@property (nonatomic, strong, nullable) id <MGLAnnotation> annotation;
```

Comment:
I can think of one case where you would need to override `setAnnotation:` until we landed an equivalent for `-mapView:didAddAnnotationViews`
(which would also improve the test case in this PR). It's possible with a workaround but e.g animating an annotation view immediately after adding an annotation is not very straight forward. You can override `setAnnotation:` to achieve that now. So I'd suggest we make this property readonly and implement `-mapView:didAddAnnotationViews`